lib/ansible/plugins/inventory/linode.py

```diff
@@ -125,18 +132,61 @@ def _filter_by_config(self, regions, types):
                 if instance.type.id in types
             ]
 
-    def _add_instances_to_groups(self):
+    def _get_instance_ip(self, instance, private=False):
+        """Returns the public or private IP address of the instance"""
+        if instance.ipv4 and len(instance.ipv4) > 0:
+            result_ip = None
+            for ip in instance.ipv4:
+                if not ipaddress.ip_address(ip).is_private and not private:
+                    result_ip = ip
+                elif ipaddress.ip_address(ip).is_private and private:
+                    result_ip = ip
+            if result_ip:
```

Comment:
Oh woops, are we missing that `raise AnsibleError...` when the `result_ip` is `None`?

Reply:
I keep forgetting private ip is optional. Added that back and a test.